vignettes/articles/writing-custom-fns.Rmd

```diff
@@ -0,0 +1,265 @@
+
+The function requires a name for the new custom check function, e.g. `"example_check"`. It then creates an `.R` script file named after the function (`example_check.R`) and saves it in the hub at the recommended location: `src/validations/R/`. The script contains basic skeleton code to create a custom check function called `example_check`.
+
+The output of `create_custom_check()` can also be parametarised through a number of arguments to include additional template code snippets (see below for examples).
+
+Let's take a look at the basic structure of a custom check function created by `create_custom_check()`:
+
+We'll start by creating a temporary "hub" for us to work in, but if you have an existing hub, you can work in there.
+
+```{r}
+hub_path <- withr::local_tempdir()
+
+create_custom_check("cstm_check_tbl_basic",
+  hub_path = hub_path
+)
+```
+
+The contents of the created file at `src/validations/R/cstm_check_tbl_basic.R` are as follows:
+
+
+```{r, code=readLines(r_path(hub_path, "cstm_check_tbl_basic")), eval=FALSE, comment=NA}
+```
+
+
+<div class="alert alert-info">
+
+### **Function naming conventions**
+
+We recommend following the naming conventions used in `hubValidations`.
+
+To distinguish `hubValidations` package functions from custom ones, we recommend prefixing custom function names with an additional prefix, e.g. `cstm_` or `cs`.
+
+- `cstm_check_file_*` for checks that operate on model output files (e.g. file location, name etc).
+- `cstm_check_tbl_*` for checks that operate on a tibble of model output data (i.e. the contents of a file).
+- `cstm_check_meta_*` for checks that operate on model metadata files.
+- `cstm_check_submission_*` for checks that operate on high level properties of the submission (e.g. timing).
+- `cstm_check_config_*` for checks that operate on the hub configuration files.
+- `cstm_check_valid_*` for checks that don't fit into the above categories.
+
+Neither of these recommendations are required for custom functions to work, but consistency is an important aspect of maintaining a hub. 
+
+
+</div>
+
+
+## Function inputs / arguments
+
+The minimum inputs required by a custom check function depend on the type of check being performed.
+
+- `file_path`: the relative path to the submission file being validated is required for all check functions. **`file_path` must therefore be included as an argument** in all custom check function.
+- `tbl` or `tbl_chr`: a tibble representation of the contents of a model output submission (with column data types matching the hub schema (`tbl`) or an all character version (`tbl_chr`) is also required by any checks that operate on the data in the submission file.
+
+Since `file_path` and `tbl` are the most common inputs to check functions, `create_custom_check()` includes them as arguments by default. This means that the custom check function will include these objects in the function call environment by default. 
+
+<div class="alert alert-warn">
+
```

Comment:
I really like this discussion and I feel like a summary heading here can help drive home the point. 

```suggestion

### Use `tbl` only with `validate_model_data()`

```